endpoints/events/event.go

```diff
@@ -173,6 +169,24 @@ func ParseEventRequest(r *http.Request) (*analytics.EventRequest, error) {
 	return event, nil
 }
 
+// Handle fetching account errors
+func HandleAccountServiceErrors(errs []error) (status int, message string) {
+	for _, er := range errs {
+		errCode := errortypes.ReadCode(er)
+
+		switch errCode {
+		case errortypes.BlacklistedAppErrorCode, errortypes.BlacklistedAcctErrorCode:
+			return http.StatusServiceUnavailable, er.Error()
+		case errortypes.AcctRequiredErrorCode:
+			return http.StatusBadRequest, er.Error()
+		default:
+			return http.StatusInternalServerError, er.Error()
```

Comment:
With a default case, the loop only runs once.

We probably want behavior like `writeError` in https://github.com/prebid/prebid-server/blob/7b59a4bd49e502b05f81624bcc908259aec757cf/endpoints/openrtb2/auction.go#L1282 or the similar but different code the other two auction endpoints.

Reply:
You're right, an overview on my part - Refactored.